Feature/filename search

d3r1w opened the pull request:
Hi @lechimp-p 
I would like to contribute a case insensitive filename search to the *diff_per_rule_bootstrap* template.

**Added:**
* diff_per_rule_bootstrap template -> case insensitive filename search

**Changed:**
* Removed some unused imports.
* Integrity checksum added for the jquery library.

chfsx commented:
Hi @lechimp-p This PR represents the changes the TB paid for. How to proceed with this? Thanks in advance!

lechimp-p commented:
Will have a look tomorrow. Thx!

lechimp-p commented on templates/diff_per_rule_bootstrap.php:
```diff
@@ -35,13 +35,52 @@ function template_diff_per_rule_bootstrap(array $report) {
         crossorigin="anonymous">
-    </script> 
+    </script>
+	<script>
```
Please reindent using spaces. I try to stick to more standard conventions in my private projects.

lechimp-p commented on templates/diff_per_rule_bootstrap.php:
```diff
@@ -35,13 +35,52 @@ function template_diff_per_rule_bootstrap(array $report) {
         crossorigin="anonymous">
     <script
         src="https://code.jquery.com/jquery-2.2.4.min.js"
+		integrity="sha384-rY/jv8mMhqDabXSo+UCggqKtdmBfd3qC2/KvyTDNQ6PcUJXaxK1tMepoQda4g5vB"
```
👍 thx

lechimp-p commented on src/Indexer/AdapterVisitor.php:
```diff
@@ -10,9 +10,6 @@
 namespace Lechimp\Dicto\Indexer;
 
-use Lechimp\Dicto\Variables\Variable;
-use PhpParser\Node as N;
```
👍 thx

lechimp-p commented on src/Indexer/ASTVisitor.php:
```diff
@@ -10,9 +10,6 @@
 
-use Lechimp\Dicto\Variables\Variable;
-use PhpParser\Node as N;
-
```
👍 thx

d3r1w commented:
@lechimp-p 
Thanks for your feedback.

- Yes this will render one search field for all rules.
- Yes it worked at my side, but I will test it again while fixing the indentation may I forgot something.
- I should have a bit time to implement the numbers.

Could you please explain the last point of your suggestions again? May I got you wrong but the resolved commits are also filtered and they are already linked to the relevant line of code.

I attached a report in case you want to have a quick look without generating everything.
[report.zip](https://github.com/lechimp-p/dicto.php/files/1077607/report.zip)

lechimp-p commented:
Currently the resolved violations are not included in the list of violations, only in the numbers. Thus, there are only red (for new violations) and gray (for violations that already existed) but no green (for resolved violations) lines in the listing of code locations.
If you also adjust the numbers according to the filter, the number of resolved violations will always have to stay the same, because there in fact are no information about where violations were resolved. These are also missing when I want to praise people on the JF.
I therefore thought: Include resolved violations (with links to code of the previous commit) as green lines in the report to be able to filter them as well.

d3r1w commented:
@lechimp-p 
Thanks for the detailed explanation. I looked only on the green bars of the rules instead of the entries ...
This should be possible but I am not entirely sure how to implement it should I add an additional query to get the resolved violations within the DiffPerRuleReport.php ?

lechimp-p commented:
I think you should add a new query [here](https://github.com/lechimp-p/dicto.php/blob/master/src/Report/Queries.php) and then just use it in the DiffPerRuleReport.

lechimp-p commented:
Or you could use `violations_of` with the id of the last run and add the information when a violation was resolved to the result of that query.

lechimp-p commented:
The first one is the cleaner solution, I think.

d3r1w commented:
@lechimp-p 
Thanks for your advice and inputs. I implemented the following:

- Resolved violations are now also shown and filtered.
- Only the visible violations are counted.
- Fixed the indentations tabs -> spaces

I tested the *diff_per_rule_bootstrap* and the *diff_per_rule* template.

I generated both the bootstrap and the text template this time.
[report.zip](https://github.com/lechimp-p/dicto.php/files/1080919/report.zip)

lechimp-p commented:
Hi Nick,

I made some further improvements, please follow my trail of commits an ask questions if you are interested.

Regards!

d3r1w commented:
@lechimp-p 
Thanks for your additional enhancements!
